core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -171,31 +170,30 @@ public void doWork() {
             log.error("Failed to add record from the Kafka topic" + topic + " the local store");
           }
         }
+
+        if (commitInterval > 0 && System.currentTimeMillis() - lastCommitTime > commitInterval) {
+          log.debug("Committing offsets");
+          lastCommitTime = System.currentTimeMillis();
+          consumer.commitSync();
+        }
       }
-    } catch (ConsumerTimeoutException cte) {
-      // Expect ConsumerTimeout == -1, so this *should* never happen
+    } catch (WakeupException we) {
+      this.consumer.close();
+    } catch (RecordTooLargeException rtle) {
       throw new IllegalStateException(
-          "KafkaStoreReaderThread's ConsumerIterator timed out despite expected infinite timeout.");
-    } catch (MessageSizeTooLargeException mstle) {
-      throw new IllegalStateException(
-          "ConsumerIterator threw MessageSizeTooLargeException. A schema has been written that "
-          + "exceeds the default maximum fetch size.", mstle);
+          "Consumer threw RecordTooLargeException. A schema has been written that "
+          + "exceeds the default maximum fetch size.", rtle);
     } catch (RuntimeException e) {
       log.error("KafkaStoreReader thread has died for an unknown reason.");
       throw new RuntimeException(e);
     }
-    
-    if (commitInterval > 0 && System.currentTimeMillis() - lastCommitTime > commitInterval) {
-      log.debug("Committing offsets");
-      consumer.commitOffsets(true);
-    }
   }
 
   @Override
   public void shutdown() {
     log.debug("Starting shutdown of KafkaStoreReaderThread.");
     if (consumer != null) {
-      consumer.shutdown();
+      consumer.wakeup();
```

Comment:
I don't think we need it because `ShutdownableThread` (which `KafkaStoreReaderThread` extends) uses a countdown latch. Although, this does mean I need to change the shutdown logic to not just call `super.shutdown()`.